sklearn/preprocessing/_encoders.py

```diff
@@ -363,6 +486,201 @@ def _compute_drop_idx(self):
                              zip(self.drop, self.categories_)],
                             dtype=np.object)
 
+    def _infrequent_enabled(self):
+        """Infrequent category is enabled."""
+        if self.max_categories is not None and self.max_categories > 1:
+            return True
+        if (isinstance(self.min_frequency, numbers.Integral)
+                and self.min_frequency > 1):
+            return True
+        if (isinstance(self.min_frequency, numbers.Real)
+                and 0.0 < self.min_frequency < 1.0):
+            return True
+        return False
+
+    def _identify_infrequent(self, category_count, n_samples, col_idx):
+        """Compute the infrequent indicies based on max_categories and
+        min_frequency.
+
+        Parameters
+        ----------
+        category_count : ndarray of shape (n_cardinality,)
+            Category counts.
+
+        n_samples : int
+            Number of samples.
+
+        col_idx : int
+            Index of the current category. Only used for the error message.
+
+        Returns
+        -------
+        output : ndarray of shape (n_infrequent_categories,) or None
+            If there are infrequent categories, indicies of infrequent
+            categories. Otherwise None.
+        """
+        # categories with no count are infrequent
+        infrequent_mask = category_count == 0
+
+        if isinstance(self.min_frequency, numbers.Integral):
+            infrequent_mask |= category_count < self.min_frequency
+        else:  # float
+            min_frequency_abs = n_samples * self.min_frequency
+            infrequent_mask |= category_count < min_frequency_abs
+
+        if (self.max_categories is not None and self.max_categories > 1
+                and self.max_categories < category_count.size):
+            # stable sort to preserve original count order
+            smallest_levels = np.argsort(
+                category_count, kind='mergesort'
+                )[:-self.max_categories + 1]
+            infrequent_mask[smallest_levels] = True
+
+        output = np.flatnonzero(infrequent_mask)
+        if output.size == category_count.size:
+            raise ValueError(f"All categories in column {col_idx} are "
+                             "infrequent")
+        return output if output.size > 0 else None
+
+    def _fit_infrequent_category_mapping(self, fit_results):
+        """Fit infrequent categories.
+
+        Defines:
```

Comment:
It wasn't clear to me what this means  when I read this. Maybe say set's the following attributes on 'self'?